### DISCUSSION TITLE
Why isn't set_interval working?

### DISCUSSION THREAD

**Discussion by djotaku** [Textualize/textual#5070] [Q&A] [answered]
My code is here: https://github.com/djotaku/sdpmpd/blob/40694248ff4b43200a875afe5dd98ef5e8eadba0/sdpmpd/tui.py

Specifically the class that starts on line 36. 

A few things: 

- I have tried the set_interval method both without explicitly putting the self.add_next_song() in the callback= and, as you can see now, explicitly calling it
- I am running this with the dev debug console so I can see the print statements. 

I commented everything out in "add next song" so that I figured I'd see 60 print statements per second. Instead I just see it once once on mount. Line 61 shows that callback = None, which seems weird to me. What am I doing wrong here? Looking at the stopwatch example, this seems like it should work.

**Comment by willmcgugan** [comment, marked as answer]
Remove the () from the callback. You want to pass in the function, without calling it.

**Reply by djotaku** [reply]
Perfect, thanks!
